### DISCUSSION TITLE
reflex-agent Errors

### DISCUSSION THREAD

**Discussion by ar3diu** [reflexsoar/reflex-api#157] [General] [answered]
Hi everyone! 

I just installed ReflexSOAR and tried to set an Elasticsearch input. Both network connectivity and DNS resolution for Elasticsearch host were successfully tested.
As (super user) credentials I tried to use both basic_auth and api_key but did not have any success with any of them.
For TLS verification - I did not set the path to CA as I do not want to verify the Elasticsearch' certificate.
I had a look at those specific lines of code indicated in the error below but couldn't figure out what I'm doing wrong.

Do you have any advice/guidance on how I can further troubleshoot this error? 

```
reflex-agent  | Traceback (most recent call last):
reflex-agent  |   File "reflex-agent.py", line 99, in <module>
reflex-agent  |     e = Elastic(i['config'], i['field_mapping'], credentials)
reflex-agent  |   File "/utils/elasticsearch.py", line 25, in __init__
reflex-agent  |     self.conn = self.build_es_connection()
reflex-agent  |   File "/utils/elasticsearch.py", line 63, in build_es_connection
reflex-agent  |     return Elasticsearch(self.config['hosts'], **es_config)
reflex-agent  | TypeError: __init__() got an unexpected keyword argument 'scheme'
```
![image](https://user-images.githubusercontent.com/61625853/167883599-d4472d7c-3b77-4efc-a70d-9745d0583ab9.png)

**Comment by n3tsurge** [comment]
Hi @anrsec can you try changing the Distro on the Input from `elasticsearch` to `opensearch` we noticed this recently as well and it appears something changed with the Elasticsearch-py package that is causing this issue and we haven't been able to pinpoint it yet

**Comment by n3tsurge** [comment, marked as answer]
I think we have pinpointed the underlying issue with using the `elasticsearch` distro we are currently testing that fix but using `opensearch` as your distro on the inputs configuration should allow you to proceed until a permanent fix is in place.

**Reply by ar3diu** [reply]
Yes, it works when I'm using `opensearch` as the value of the `distro` field. It never occurred to me to try this way.

Thanks a lot!

**Comment by Grok42** [comment]
Just curious... Haven't spun up an Elasticsearch instance yet (Using Opensearch).

Noticed this in [reflex-api/app/services/threat_list_poller](https://github.com/reflexsoar/reflex-api/blob/main/app/services/threat_list_poller/base.py)

`def build_elastic_connection(self):`
` . . . `
`    elif self.app.config['ELASTICSEARCH_AUTH_SCHEMA'] == 'api':`
`    elastic_connection['api_key'] = (username,password)`

**Is 'username' (here and elsewhere) stored as 'admin' or user supplied?**

Saw this in Elastic Credentials API for both 8.0+ and 7.17

[Credentials API](https://www.elastic.co/guide/en/app-search/current/credentials.html)
> 
>Authentication
>
>The credentials endpoint requires a Private Admin Key.
>
>A Private Admin Key is not generated for you upon account creation, so you must create one.
>
>The key begins with admin- and is required within all of your requests.

**Reply by n3tsurge** [reply]
It's this https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html.  in the case of using API auth the username would be the ID and password would be the key

**Reply by Grok42** [reply]
Ah, I thought "admin" might be a Elasticsearch system account (one where you can change password, but "admin" user stays the same).  Have seen references to _id , but this linked page on Credentials API seemed explicit to "admin-key"
